### DISCUSSION TITLE
Create custom Field Type

### DISCUSSION THREAD

**Discussion by ferryhtw** [pocketbase/pocketbase#1754] [Q&A]
Is it possible to create custom Field Type with Pocketbase Go frameworks?
If this possible do I know where to start (such as hook up or documentation to read) ?

**Comment by ganigeorgiev** [comment]
This is not possible at the moment, mostly because it will require some dynamic changes in Admin UI and currently the admin cannot be extended (_this will be handled in the future as part of #898_).

Feel free to let me know what field you are missing and eventually we may integrate it in the main package as long as it is generic and not too user-land code specific.

**Reply by ferryhtw** [reply]
Hi @ganigeorgiev , 
First I want to say thank you for working on this amazing works and I fall in love immidiately after I watch it on youtube. 
There are two custom field that I planned/wish to for my shop/store inventory db:
1. Auto Populate Field for Sequence - Useful if you have a store stock Product Codes or Discount Category eg. DR-01 DR-02 is for dairy product codes or AP-MP-0001 - AP-MP-0002 for non perishable - Meat Product stock . 
2. Auto Populate Field based on another field - 
   example: 
        everytime I create a new Item with Fields 
               StoreId: C
               RackId: D
               StoryId: 3
       Folowing The field AssetID will be auto populated : CD3-0001 which is a combination of StoreID RackID StoryID "dash character" and a auto sequence field number which is auto pupulated by feature point 1 above. 

Right now, What in my mind is to utilizing "OnRecordBeforeCreateRequest" event hook. 
However, t would be nice if can add this as field type during a runtime, so my colleague can do it them self evertime they need to add new numbering pattern.
